internal/encryption/encryption.go

```diff
@@ -56,88 +62,327 @@ func (n NoOpEncryptor) DecryptJSON(input json.RawMessage, output any) error {
 	Encrypted []byte `json:"encrypted"`
 }
 
-func (e Encryptor) EncryptJSON(input any) (json.RawMessage, error) {
+func (e DeprecatedEncryptor) EncryptJSON(input any) (json.RawMessage, error) {
 	msg, err := json.Marshal(input)
 	if err != nil {
 		return nil, fmt.Errorf("failed to marshal input: %w", err)
 	}
 
+	encrypted, err := encryptBytesForStreaming(e.primitive, msg)
+	if err != nil {
+		return nil, fmt.Errorf("failed to encrypt data: %w", err)
+	}
+
+	out, err := json.Marshal(DeprecatedEncryptedPayload{Encrypted: encrypted})
+	if err != nil {
+		return nil, fmt.Errorf("failed to marshal encrypted data: %w", err)
+	}
+	return out, nil
+}
+
+func (e DeprecatedEncryptor) DecryptJSON(input json.RawMessage, output any) error {
+	var payload DeprecatedEncryptedPayload
+	if err := json.Unmarshal(input, &payload); err != nil {
+		return fmt.Errorf("failed to unmarshal encrypted payload: %w", err)
+	}
+
+	decryptedBuffer, err := decryptBytesForStreaming(e.primitive, payload.Encrypted)
+	if err != nil {
+		return fmt.Errorf("failed to decrypt data: %w", err)
+	}
+
+	if err := json.Unmarshal(decryptedBuffer, output); err != nil {
+		return fmt.Errorf("failed to unmarshal decrypted data: %w", err)
+	}
+
+	return nil
+}
+
+func encryptBytesForStreaming(streamingPrimitive tink.StreamingAEAD, clearText []byte) ([]byte, error) {
 	encryptedBuffer := &bytes.Buffer{}
-	msgBuffer := bytes.NewBuffer(msg)
-	writer, err := e.primitive.NewEncryptingWriter(encryptedBuffer, nil)
+	msgBuffer := bytes.NewBuffer(clearText)
+	writer, err := streamingPrimitive.NewEncryptingWriter(encryptedBuffer, nil)
 	if err != nil {
 		return nil, fmt.Errorf("failed to create encrypting writer: %w", err)
 	}
-
 	if _, err := io.Copy(writer, msgBuffer); err != nil {
 		return nil, fmt.Errorf("failed to copy encrypted data: %w", err)
 	}
 	if err := writer.Close(); err != nil {
 		return nil, fmt.Errorf("failed to close encrypted writer: %w", err)
 	}
 
-	out, err := json.Marshal(EncryptedPayload{Encrypted: encryptedBuffer.Bytes()})
+	return encryptedBuffer.Bytes(), nil
+}
+
+func decryptBytesForStreaming(streamingPrimitive tink.StreamingAEAD, encrypted []byte) ([]byte, error) {
+	encryptedBuffer := bytes.NewReader(encrypted)
+	decryptedBuffer := &bytes.Buffer{}
+	reader, err := streamingPrimitive.NewDecryptingReader(encryptedBuffer, nil)
 	if err != nil {
-		return nil, fmt.Errorf("failed to marshal encrypted data: %w", err)
+		return nil, fmt.Errorf("failed to create decrypting reader: %w", err)
 	}
-	return out, nil
+	if _, err := io.Copy(decryptedBuffer, reader); err != nil {
+		return nil, fmt.Errorf("failed to copy decrypted data: %w", err)
+	}
+	return decryptedBuffer.Bytes(), nil
 }
 
-func (e Encryptor) DecryptJSON(input json.RawMessage, output any) error {
-	var payload EncryptedPayload
-	if err := json.Unmarshal(input, &payload); err != nil {
-		return fmt.Errorf("failed to unmarshal encrypted payload: %w", err)
+type SubKey string
+
+const (
+	Logs  SubKey = "logs"
```

Comment:
I think add a `SubKey` suffix to these, just to be clear when browsing the symbols.